fastfield_codecs/src/multilinearinterpol.rs

```diff
@@ -315,39 +315,35 @@ impl FastFieldCodecSerializer for MultiLinearInterpolFastFieldSerializer {
     /// where the local maxima are for the deviation of the calculated value and
     /// the offset is also unknown.
     fn estimate(fastfield_accessor: &impl FastFieldDataAccess, stats: FastFieldStats) -> f32 {
-        let first_val_in_first_block = fastfield_accessor.get_val(0);
-        let last_elem_in_first_chunk = CHUNK_SIZE.min(stats.num_vals);
-        let last_val_in_first_block =
-            fastfield_accessor.get_val(last_elem_in_first_chunk as u64 - 1);
+        let first_chunk: Vec<_> = (0..CHUNK_SIZE)
+            .map(|idx| fastfield_accessor.get_val(idx)) // todo replace with iterator, when
```

Comment:
there is a bug if `num_vals < CHUNK_SIZE`

Reply:
That can't happen, the codec is only enabled for vals > 5000

Reply:
I added an assert for it